src/components/MarketMySharesForm/index.js

```diff
@@ -61,11 +67,11 @@ class MarketMySharesForm extends Component {
     const { extendedSellId } = this.state
     const { selectedSellAmount, marketShares, initialize } = this.props
     const sellAmountAndMarketSharesAreDefined =
-      selectedSellAmount === undefined && extendedSellId !== undefined && marketShares.length
+      selectedSellAmount === undefined && extendedSellId !== undefined && Object.keys(marketShares).length
```

Comment:
We are using in many places ```.length``` for calculating if variables have data. I think when we do the refactoring we should extract this function somewhere else.